dev/aws/iam.ts

```diff
@@ -1,3 +1,958 @@
+  "lakeformation.amazonaws.com",
+  "lambda.amazonaws.com",
+  "lex.amazonaws.com",
+  "license-manager.amazonaws.com",
+  "lightsail.amazonaws.com",
+  "logger.cloudfront.amazonaws.com",
+  "logs.amazonaws.com",
+  "machinelearning.amazonaws.com",
+  "macie.amazonaws.com",
+  "managedservices.amazonaws.com",
+  "mediaconnect.amazonaws.com",
+  "mediaconvert.amazonaws.com",
+  "mediapackage.amazonaws.com",
+  "mediastore.amazonaws.com",
+  "mediatailor.amazonaws.com",
+  "member.org.stacksets.cloudformation.amazonaws.com",
+  "metering-marketplace.amazonaws.com",
+  "migrationhub.amazonaws.com",
+  "mobilehub.amazonaws.com",
+  "monitoring.amazonaws.com",
+  "monitoring.rds.amazonaws.com",
+  "ops.apigateway.amazonaws.com",
+  "opsworks-cm.amazonaws.com",
+  "opsworks.amazonaws.com",
+  "organizations.amazonaws.com",
+  "pinpoint.amazonaws.com",
+  "polly.amazonaws.com",
+  "qldb.amazonaws.com",
+  "quicksight.amazonaws.com",
+  "ram.amazonaws.com",
+  "rds.amazonaws.com",
+  "redshift.amazonaws.com",
+  "rekognition.amazonaws.com",
+  "replication.dynamodb.amazonaws.com",
+  "replicator.lambda.amazonaws.com",
+  "resource-groups.amazonaws.com",
+  "robomaker.amazonaws.com",
+  "route53.amazonaws.com",
+  "route53domains.amazonaws.com",
+  "route53resolver.amazonaws.com",
+  "s3.amazonaws.com",
+  "sagemaker.amazonaws.com",
+  "secretsmanager.amazonaws.com",
+  "securityhub.amazonaws.com",
+  "serverlessrepo.amazonaws.com",
+  "servicecatalog.amazonaws.com",
+  "servicediscovery.amazonaws.com",
+  "ses.amazonaws.com",
+  "shield.amazonaws.com",
+  "signer.amazonaws.com",
+  "signin.amazonaws.com",
+  "sms.amazonaws.com",
+  "sns.amazonaws.com",
+  "spotfleet.amazonaws.com",
+  "sqs.amazonaws.com",
+  "ssm.amazonaws.com",
+  "sso.amazonaws.com",
+  "states.amazonaws.com",
+  "storagegateway.amazonaws.com",
+  "sts.amazonaws.com",
+  "support.amazonaws.com",
+  "swf.amazonaws.com",
+  "tagging.amazonaws.com",
+  "tagpolicies.tag.amazonaws.com",
+  "transcribe.amazonaws.com",
+  "transfer.amazonaws.com",
+  "translate.amazonaws.com",
+  "trustedadvisor.amazonaws.com",
+  "tts.amazonaws.com",
+  "vmie.amazonaws.com",
+  "waf-regional.amazonaws.com",
+  "waf.amazonaws.com",
+  "workdocs.amazonaws.com",
+  "worklink.amazonaws.com",
+  "workmail.amazonaws.com",
+  "workspaces.amazonaws.com",
+  "xray.amazonaws.com",
+];
+
+const identityStruct = [
+  { command: "aws iam list-users", parentKey: "Users", childKey: "Arn" },
+  { command: "aws iam list-groups", parentKey: "Groups", childKey: "Arn" },
+  { command: "aws iam list-roles", parentKey: "Roles", childKey: "Arn" },
+];
+
+const _prefixFile = "file://";
+
+const appendFolderPath = (tokens: string[], prefix: string): string => {
+  const baseLSCommand = "\\ls -1ApL ";
+  let whatHasUserTyped = tokens[tokens.length - 1];
+
+  if (whatHasUserTyped.startsWith(prefix)) {
+    whatHasUserTyped = whatHasUserTyped.slice(prefix.length);
+  } else {
+    return `echo '${prefix}'`;
+  }
```

Comment:
Just a tiny change, but reverting the if allows us to omit  the else block and make the code more readable imho
```suggestion
  if (!whatHasUserTyped.startsWith(prefix)) {
     return `echo '${prefix}'`;
  } 
  whatHasUserTyped = whatHasUserTyped.slice(prefix.length);
```